Add agent-executable workflow docs for CVE fixes

dfarrell07 opened the pull request:
CVE workflow docs and commit templates for AI agents and human devs.

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

dfarrell07 committed:
Add agent-executable workflow docs for CVE fixes

CVE workflow docs and commit templates for AI agents and human devs.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
The CVE workflow is able to one-shot fixes like https://github.com/submariner-io/submariner/pull/3677.

skitt reviewed (changes requested):
See comments in https://github.com/submariner-io/lighthouse/pull/1928

tpantelis commented on .agents/workflows/cve-fix.md:
````diff
@@ -0,0 +1,231 @@
+
+Fetch latest changes:
+```bash
+git fetch
````
I assume this is done in the project directory. Perhaps we should explicitly state this?

tpantelis commented on .agents/workflows/cve-fix.md:
````diff
@@ -0,0 +1,231 @@
+make pkg/natdiscovery/proto/natdiscovery.pb.go
+git add -u
+git diff -I'^//' --staged
+```
````
This isn't related to CVE's and would only be relevant if someone happened to have changed _natdiscovery.pb.go_ or the protobuf dependency was bumped (both of which are uncommon). It doesn't seem necessary to explain here and removing would simplify the doc but If you want to keep it then I think it would be useful to explain the step and why it may be needed. 

Also, if _natdiscovery.pb.go_ already exists, then it needs to be removed before doing `make`.

dfarrell07 committed:
Enable stdlib CVE detection in workflow

Grype detects stdlib CVEs from compiled binaries, not go.mod.
Add BUILD_UPX=false builds to preserve Go version for scanning.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
If https://github.com/submariner-io/submariner/pull/3678/commits/ee4859b6beb33d33905e65c139de6b112817013d addresses https://github.com/submariner-io/submariner/pull/3695#pullrequestreview-3411608907 I'll spread it to the other WIP CVE workflows.
